### PR TITLE
app-editors/nvi: various fixes

### PR DESCRIPTION
**Pull request by doughdemon**
@proxy-maint

### CODE REVIEW

File: app-editors/nvi/nvi-1.81.6-r6.ebuild [hunk 1]
```diff
@@ -43,22 +43,22 @@ RDEPEND="${CDEPEND}
...
 src_prepare() {
-
-	epatch "${FILESDIR}"/${P}-strlen-macro-renaming.patch
-	epatch "${FILESDIR}"/${P}-db44.patch
-	epatch "${FILESDIR}"/${P}-db.patch
-	epatch "${FILESDIR}"/${P}-perl-as-needed.patch
-	epatch "${FILESDIR}"/${P}-perl-shortnames.patch
-	epatch "${FILESDIR}"/${P}-ac_config_header.patch
-	epatch "${FILESDIR}"/${P}-use_pkgconfig_for_ncurses.patch
+	eapply "${FILESDIR}"/${P}-strlen-macro-renaming.patch
```
Comment (SoapGentoo) [inline app-editors/nvi/nvi-1.81.6-r6.ebuild]: don't use bajillions of `eapply`'s, just drop them into a bash array called `PATCHES` and then call default instead of `eapply` + `eapply_user`

File: app-editors/nvi/nvi-1.81.6-r6.ebuild [hunk 2]
```diff
@@ -43,22 +43,22 @@ RDEPEND="${CDEPEND}
...
+	eapply_user
 
 	cd dist || die
 	chmod +x findconfig || die
 
-	append-cppflags -I"$(db_includedir ${DBSLOTS})"
-
-	sed -i -e "s@-ldb@-l$(db_libname ${DBSLOTS})@" configure.in || die
-	rm -f configure || die
+	mv configure.in configure.ac || die
```
Comment (SoapGentoo) [inline app-editors/nvi/nvi-1.81.6-r6.ebuild]: shorten this to
`mv configure.{in,ac} || die`

File: app-editors/nvi/nvi-1.81.6-r6.ebuild [hunk 3]
```diff
@@ -70,24 +70,24 @@ src_configure() {
 	use tcl && ! use unicode && myconf="${myconf} --enable-tclinterp"
 
 	append-cppflags '-D_PATH_MSGCAT="\"/usr/share/vi/catalog/\""'
+	append-cppflags -I"$(db_includedir ${DBSLOTS})"
 
 	pushd dist 2>/dev/null
```
Comment (SoapGentoo) [inline app-editors/nvi/nvi-1.81.6-r6.ebuild]: missing `|| die`

File: app-editors/nvi/nvi-1.81.6-r6.ebuild [hunk 4]
```diff
@@ -70,24 +70,24 @@ src_configure() {
...
 	econf \
 		--program-prefix=n \
-		${myconf} \
-		|| die "configure failed"
+		${myconf}
 	popd 2>/dev/null
```
Comment (SoapGentoo) [inline app-editors/nvi/nvi-1.81.6-r6.ebuild]: missing `|| die`

File: app-editors/nvi/nvi-1.81.6-r6.ebuild [hunk 5]
```diff
@@ -70,24 +70,24 @@ src_configure() {
...
 }
 
 src_compile() {
 	pushd dist 2>/dev/null
```
Comment (SoapGentoo) [inline app-editors/nvi/nvi-1.81.6-r6.ebuild]: missing `|| die`

File: app-editors/nvi/nvi-1.81.6-r6.ebuild [hunk 6]
```diff
@@ -70,24 +70,24 @@ src_configure() {
...
-	emake || die "make failed"
+	emake
```
Comment (SoapGentoo) [inline app-editors/nvi/nvi-1.81.6-r6.ebuild]: have you tried `default` instead of `emake`?

File: app-editors/nvi/nvi-1.81.6-r6.ebuild [hunk 7]
```diff
@@ -70,24 +70,24 @@ src_configure() {
...
 	popd 2>/dev/null
```
Comment (SoapGentoo) [inline app-editors/nvi/nvi-1.81.6-r6.ebuild]: missing `|| die`

File: app-editors/nvi/nvi-1.81.6-r6.ebuild [hunk 8]
```diff
@@ -70,24 +70,24 @@ src_configure() {
...
 }
 
 src_install() {
 	pushd dist 2>/dev/null
```
Comment (SoapGentoo) [inline app-editors/nvi/nvi-1.81.6-r6.ebuild]: missing `|| die`

File: app-editors/nvi/nvi-1.81.6-r6.ebuild [hunk 9]
```diff
@@ -70,24 +70,24 @@ src_configure() {
...
-	emake -j1 DESTDIR="${D}" install || die "install failed"
+	emake DESTDIR="${D}" install
```
Comment (SoapGentoo) [inline app-editors/nvi/nvi-1.81.6-r6.ebuild]: have you tried `default` instead of `emake`?

File: app-editors/nvi/nvi-1.81.6-r6.ebuild [hunk 10]
```diff
@@ -70,24 +70,24 @@ src_configure() {
...
 	popd 2>/dev/null
```
Comment (SoapGentoo) [inline app-editors/nvi/nvi-1.81.6-r6.ebuild]: missing `|| die`

**Comment by doughdemon** [comment]
Thanks for the review!
A new commit is pushed with all issues addressed.

File: app-editors/nvi/nvi-1.81.6-r6.ebuild [hunk 11]
```diff
@@ -70,25 +73,25 @@ src_configure() {
...
+	popd 2>/dev/null || die
 }
 
 src_install() {
-	pushd dist 2>/dev/null
-	emake -j1 DESTDIR="${D}" install || die "install failed"
-	popd 2>/dev/null
+	pushd dist 2>/dev/null || die
+	default
+	popd 2>/dev/null || die
```
Comment (mgorny) [inline app-editors/nvi/nvi-1.81.6-r6.ebuild]: Likewise.

**Comment by mgorny** [comment]
@gentoo/proxy-maint
